dom.go

```diff
@@ -594,7 +595,7 @@ func (h *HTML) insertBefore(node jsObject, child *HTML) {
 type List []ComponentOrHTML
 
 // WithKey wraps the List in a Keyer using the given key.
-func (l List) WithKey(key interface{}) MarkupOrChild {
+func (l List) WithKey(key interface{}) ComponentOrHTML {
```

Comment:
Curious to know why you chose `MarkupOrChild` initially.. maybe I am overlooking something. But it seems like `List.WithKey` should be usable inside `vecty.If` just the same as a `List` can.

Reply:
I think only because it was discussed in the issue when we were planning this, this change is fine.